windows-gaming-driver/src/driver/clipboard.rs

```diff
@@ -184,9 +180,9 @@ impl X11Clipboard {
             self.connection.flush();
 
             Ok(())
-        }).map_err(|()| unimplemented!())).then(|_| Ok(())));
+        });
 
-        ClipboardContext { listener, lost_recv, query_recv, data_recv, resp_send }
+        Box::new(xcb_listener.join(responder.map_err(|()| unimplemented!())).then(|_| Ok(())))
```

Comment:
If that's the case, couldn't you put `unreachable!` there? It'll have the same effect but without implying a TODO and instead showing that it can't fail.